pgx-tests/src/tests/schema_tests.rs

```diff
@@ -4,14 +4,66 @@ use pgx::*;
 
 #[pgx::pg_schema]
 mod test_schema {
+    use pgx::datum::sql_entity_graph::{PgxSql, SqlGraphEntity};
     use pgx::*;
     use serde::{Deserialize, Serialize};
 
     #[pg_extern]
     fn func_in_diff_schema() {}
 
+    #[pg_extern(sql = false)]
+    fn func_elided_from_schema() {}
+
+    #[pg_extern(sql = generate_function)]
+    fn func_generated_with_custom_sql() {}
+
     #[derive(Debug, PostgresType, Serialize, Deserialize)]
     pub struct TestType(pub u64);
+
+    #[derive(Debug, PostgresType, Serialize, Deserialize)]
+    #[pgx(sql = false)]
+    pub struct ElidedType(pub u64);
+
+    #[derive(Debug, PostgresType, Serialize, Deserialize)]
+    #[pgx(sql = generate_type)]
+    pub struct OtherType(pub u64);
+
+    #[derive(Debug, PostgresType, Serialize, Deserialize)]
+    #[pgx(sql = "CREATE TYPE test_schema.ManuallyRenderedType;")]
+    pub struct OverriddenType(pub u64);
+
+    fn generate_function(
+        entity: &SqlGraphEntity,
+        _context: &PgxSql,
+    ) -> Result<String, Box<dyn std::error::Error + Send + Sync + 'static>> {
+        if let SqlGraphEntity::Function(ref func) = entity {
+            Ok(format!("\
+                CREATE OR REPLACE FUNCTION test_schema.\"func_generated_with_custom_name\"() RETURNS void\n\
+                LANGUAGE c /* Rust */\n\
+                AS 'MODULE_PATHNAME', '{unaliased_name}_wrapper';\
+                ",
+                unaliased_name = func.unaliased_name,
```

Comment:
😅 I'm gonna need to clean up this API a bit.

Reply:
I had a similar thought when I was thinking through how to expose stuff to those functions haha. In my perfect world, users of this would get access to a hypothetical `SqlBuilder` API that let you replicate this using the builder pattern:

```rust
fn generate_function(entity: &SqlGraphEntity, builder: &mut SqlBuilder, _context: &PgxSql) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
    if let SqlGraphEntity::Function(ref f) = entity {
        // get_or_create lets you modify existing entities after initial construction, but before the builder is consumed
        // returns a entity builder that holds a mutable reference to the parent builder
        // when the entity is finalized, only then is the entity added to the parent builder
        let mut func = builder.get_or_create_function(f.name);
        func
          .args(&[]) // defaults to no arguments if not set
          .returns("void") // defaults to 'void' if not set
          .replace(true) // indicates the builder should use CREATE OR REPLACE
          .lang(FunctionLanguage::Rust) // lowers to LANGUAGE c /* Rust */
          .as_nif("MODULE_PATHNAME", format!("{}_wrapper", func.unaliased_name)) // indicates the function is a Natively Implemented Function from the given module
          .build()?; // adds the built function to the parent builder
        Ok(())
    } else {
        panic!("expected extern function entity, got {:?}", entity);
    }
}
```

It'd be a big undertaking to cover all possible entities, but could probably start by just covering the cases that pgx itself requires, expose some kind of escape hatch akin to `extension_sql!`, and expand the API with specific entities as needed. Not sure if it is worth it or not, but would be a lot nicer and less error prone to work with!